opengever/maintenance/scripts/sql_contact_migration.py

```diff
@@ -0,0 +1,110 @@
+"""
+This script is used to export sql contacts in to bundle and removes contact
+references in manual journal entries.
+
+bin/instance run ./scripts/sql_contact_migration.py
+"""
+
+from opengever.contact.models.org_role import OrgRole
+from opengever.contact.models.organization import Organization
+from opengever.contact.models.person import Person
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_option_parser
+from opengever.maintenance.debughelpers import setup_plone
+from uuid import uuid4
+import json
+import logging
+import os
+import sys
+import time
+import transaction
+
+logger = logging.getLogger('opengever.maintenance')
+handler = logging.StreamHandler(stream=sys.stdout)
+handler.setFormatter(logging.Formatter('%(levelname)s %(message)s'))
+logging.root.addHandler(handler)
+logging.root.setLevel(logging.INFO)
+
+
+class SqlContactExporter(object):
+
+    def __init__(self, bundle_directory):
+        self.bundle_directory = bundle_directory
+        self.contact_mapping = {}
+        self.org_role_mapping = {}
+
+    def run(self):
+        os.mkdir(self.bundle_directory)
+
+        self.export()
+
+    def export(self):
+        persons = list(self.get_persons())
+        self.export_json('people.json', persons)
+
+        organizations = list(self.get_organizations())
+        self.export_json('organizations.json', organizations)
+
+        org_roles = list(self.get_org_roles())
+        self.export_json('memberships.json', org_roles)
+
+    def get_persons(self):
+        for person in Person.query:
+            kub_uid = str(uuid4())
+            self.contact_mapping[person.contact_id] = kub_uid
+
+            yield {
+                'id': kub_uid,
+                'third_party_id': person.former_contact_id,
```

Comment:
That's what I thought at the beginning. But the mercury team has changed the procedure a little bit. Because we do not store all the contact metadata in our GEVER sql contacts, they decided to use the "Scope-Partner" database as the "main source" for the migration. The following procedure is now planned:

**1. Step**
- Export of all GEVER SQL contacts (but only with a minimal set of Metadata) in to a KUB Bundle. GEVER creates the generates the `kub_id` for each item
- Migration of the participations

**2. Step**
- Export of all ScopePartner contacts in to a KUB Bundle, with a full set of metadata.

**3. Step**
- Merge these two KUB bundles using the ScopePartner ID and import it to KUB

Means GEVER only provides a minimal set of data and the common ID, need to be the ScopePartner (`former_contact_id` in GEVER). Because ScopePartner has no reference to gevers `contact_id`.

Reply:
ah ok I see. How about contacts that were created in Gever and never were in ScopePartner? Do all contacts in Gever have a `former_contact_id`?

Reply:
There was never a possibility for our users to create contacts manually. So yes all contacts have an `former_contact_id`